saleor/plugins/sendgrid/tests/test_plugin.py

```diff
@@ -14,6 +19,134 @@ def test_get_event_map():
         assert event in EVENT_MAP
 
 
+external_notification_trigger_query = """
+      mutation ExternalNotificationTrigger(
+        $input: ExternalNotificationTriggerInput!
+        $pluginId: String
+      ) {
+          externalNotificationTrigger(
+            input: $input,
+            pluginId: $pluginId
+          ) {
+            errors {
+              message
+            }
+          }
+      }
+    """
+
+
+@patch("saleor.plugins.sendgrid.plugin.EVENT_MAP")
+def test_notify_via_external_notification_trigger_with_extra_payload(
+    mocked_event_map,
+    staff_users,
+    sendgrid_email_plugin,
+):
+
+    test_json = {"TEST": "VALUE", "TEST_LIST": ["GUEST1", "GUEST2"]}
+    plugin = sendgrid_email_plugin(
+        active=True, api_key="AB12", account_password_reset_template_id="123"
+    )
+    mocked_event_task = MagicMock()
+    event_map = {
+        UserNotifyEvent.ACCOUNT_PASSWORD_RESET: (
+            mocked_event_task,
+            "account_password_reset_template_id",
+        )
+    }
+    mocked_event_map.__getitem__.side_effect = event_map.__getitem__
+    mocked_event_map.__contains__.return_value = True
+    mocked_event_map.get.side_effect = event_map.get
+
+    expected_payload = [get_user_custom_payload(user) for user in staff_users]
+    for payload in expected_payload:
+        payload["extra_payload"] = test_json
+
+    plugin.notify(UserNotifyEvent.ACCOUNT_PASSWORD_RESET, expected_payload, None)
+
+    mocked_event_task.delay.assert_called_once_with(
+        expected_payload, asdict(plugin.config)
+    )
+
+
+@patch("sendgrid.base_interface.BaseInterface.send")
+def test_send_notification_to_customers_with_product_variant_payload(
+    notify_single_plugin_mock,
+    settings,
+    staff_users,
+    staff_api_client,
+    permission_manage_users,
+    sendgrid_email_plugin,
+    product_with_single_variant,
+):
+
+    settings.PLUGINS = [
```

Comment:
You're testing SendgridPlugin and activating PluginSample. I think that this settings is not needed here